docs/iris/src/developers_guide/release.rst

```diff
@@ -0,0 +1,66 @@
+Release
+*******
+
+A release of Iris is a tag on github in the SciTools organisation master.
+
+
+Release Branch
```

Comment:
i think this is further expanding the scope of the page, into how we decide when we are ready for a release.  I don't think this needs documenting here, we can't write down everything we do, and this doesn't have anything in it which is a step we always want to do the same for each release.  

I don't mind changing my mind, but for now I've left this out.